constrain-lib/src/main/resources/schema/dependency-constraints-schema.json

```diff
@@ -0,0 +1,137 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "type": "object",
+  "title": "Dependency Constraints Schema.",
+  "description": "Defines a set of constraints to be applied to a Gradle Build. When used as a Gradle Plugin, constraints in this file will constrain the project buildscript and project dependencies. When leveraged as a Gradle feature, these constraints apply to all configurations in all scopes. \n\n FORMATTING RULES:\n In addition to this schema, there are formatting requirements that can not be enforced by JSON schema.\n Rules:\n \t- All indentation MUST use two spaces.\n \t- All keys MUST have the colon `:` directly after the key name.\n \t- All non-map and non-object values MUST be on the same line as the key.\n \t- The `dependencyConstraints` array must be sorted lexicographically. See the description on `#/$defs/dependencyConstraints` for more information.\n \t- The file MUST be terminated with a newline.\n These formatting rules are enforced by Gradle when the file is loaded. The goal is to ensure that automated tooling (eg. Dependabot and Rennotate) will generate a consistent file that won't cause annoying white space diffs. ",
```

Comment:
This is way more readable in the YAML version of the file.

https://github.com/gradle/gradle-dependency-constrain/pull/1/files#diff-26eaa682062e2219e4fc20dde65a2a6901518dc8ca66268eb1638cdfce444e97R7-R21

Reply:
Do we plan to support YAML? You can treat every json as a YAML file without modifications, but not vice-versa.

Reply:
No, I'm writing the spec in YAML and converting it to JSON. There is only one file format, JSON